File: app/src/main/java/com/github/anrimian/musicplayer/ui/main/external_player/ExternalPlayerView.java
package com.github.anrimian.musicplayer.ui.main.external_player;

import com.github.anrimian.musicplayer.data.models.composition.source.ExternalCompositionSource;
import com.github.anrimian.musicplayer.ui.common.error.ErrorCommand;

import javax.annotation.Nullable;

import moxy.MvpView;
import moxy.viewstate.strategy.alias.AddToEndSingle;
public interface ExternalPlayerView extends MvpView {

    @AddToEndSingle
    void showPlayerState(boolean isPlaying);

    @AddToEndSingle
    void displayComposition(ExternalCompositionSource source);

    @AddToEndSingle
    void showTrackState(long currentPosition, long duration);

    @AddToEndSingle
    void showRepeatMode(int mode);

    @AddToEndSingle
    void showPlayErrorState(@Nullable ErrorCommand errorCommand);

    @AddToEndSingle
    void showKeepPlayerInBackground(boolean externalPlayerKeepInBackground);

    @AddToEndSingle
    void displayPlaybackSpeed(float speed);

    @AddToEndSingle
    void showSpeedChangeFeatureVisible(boolean visible);
}

File: app/src/main/java/com/github/anrimian/musicplayer/ui/main/external_player/ExternalPlayerView.kt
package com.github.anrimian.musicplayer.ui.main.external_player

import com.github.anrimian.musicplayer.data.models.composition.source.ExternalCompositionSource
import com.github.anrimian.musicplayer.ui.common.error.ErrorCommand
import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle

interface ExternalPlayerView : MvpView {

    @AddToEndSingle
    fun showPlayerState(isPlaying: Boolean)

    @AddToEndSingle
    fun displayComposition(source: ExternalCompositionSource)

    @AddToEndSingle
    fun showTrackState(currentPosition: Long, duration: Long)

    @AddToEndSingle
    fun showRepeatMode(mode: Int)

    @AddToEndSingle
    fun showPlayErrorState(errorCommand: ErrorCommand?)

    @AddToEndSingle
    fun showKeepPlayerInBackground(externalPlayerKeepInBackground: Boolean)

    @AddToEndSingle
    fun displayPlaybackSpeed(speed: Float)

    @AddToEndSingle
    fun showSpeedChangeFeatureVisible(visible: Boolean)

}